application/app/controllers/download_files_controller.rb

```diff
@@ -11,14 +12,24 @@ def cancel
       return redirect_back fallback_location: root_path, alert: t('download_files.file_not_found', file_id: file_id, project_id: project_id)
     end
 
-    if file.status.downloading?
+    was_downloading = file.status.downloading?
+    if was_downloading
       command_client = Command::CommandClient.new(socket_path: ::Configuration.command_server_socket_file)
       request = Command::Request.new(command: 'download.cancel', body: {project_id: project_id, file_id: file_id})
       response = command_client.request(request)
       return redirect_back fallback_location: root_path, alert: t('download_files.file_cancellation_error', filename: file.filename) if response.status != 200
     end
 
     if file.update(status: FileStatus::CANCELLED)
+      unless was_downloading
+        log_event(
```

Comment:
This is better placed in the EventLogger.
I think, similar to log_project_event, there should be a log_download_event(download_file, ....)

Reply:
I moved it to the module